tools/autograd/gen_inplace_or_view_type.py

```diff
@@ -301,9 +308,8 @@ def emit_view_body(fn: NativeFunctionWithDifferentiabilityInfo, var: str) -> Tup
         else:
             _, unpacked_bindings = unpack_args(f)
             call += emit_view_lambda(f, unpacked_bindings)
-            creation_meta = ('at::GradMode::is_enabled() ? '
-                             'CreationMeta::DEFAULT : '
-                             'CreationMeta::NO_GRAD_MODE')
+            creation_meta = ('(!at::GradMode::is_enabled() || InferenceMode::is_enabled()) ? '
```

Comment:
I guess this raises the question again whether or not InferenceMode should also toggle `at::GradMode::is_enabled`. Doing so the easy way will cause an extra TLS change in the InferenceMode guard (although that isn't really a big deal).